protocol/contracts/dao/Comptroller.sol

```diff
@@ -71,37 +71,35 @@ contract Comptroller is Setters {
     }
 
     function increaseSupply(uint256 newSupply) internal returns (uint256, uint256, uint256) {
-        (uint256 newRedeemable, uint256 lessDebt) = (0, 0);
+        (uint256 newRedeemable, uint256 lessDebt, uint256 poolReward) = (0, 0, 0);
 
         // 1. True up redeemable pool
         uint256 totalRedeemable = totalRedeemable();
         uint256 totalCoupons = totalCoupons();
         if (totalRedeemable < totalCoupons) {
-            newRedeemable = totalCoupons.sub(totalRedeemable);
-            newRedeemable = newRedeemable > newSupply ? newSupply : newRedeemable;
-            mintToRedeemable(newRedeemable);
 
-            newSupply = newSupply.sub(newRedeemable);
-        }
+            uint256 newPotentialRedeemable = totalCoupons.sub(totalRedeemable);
+            newPotentialRedeemable = newPotentialRedeemable > newSupply ? newSupply : newPotentialRedeemable;
 
-        // 2. Eliminate debt
-        uint256 totalDebt = totalDebt();
-        if (newSupply > 0 && totalDebt > 0) {
-            lessDebt = totalDebt > newSupply ? newSupply : totalDebt;
-            decreaseDebt(lessDebt);
 
-            newSupply = newSupply.sub(lessDebt);
+            poolReward = newPotentialRedeemable.mul(Constants.getOraclePoolRatio()).div(100);
```

Comment:
This 20% I think needs to be higher up, for example:

newSupply -> 200, coupons -> 100 will do the following:

newPotentialRedeemable -> 100
poolReward -> 20
newRedeemable -> 80
newBonded -> 100

which results in there still being 20 coupons leftover at the end.

This case should rather be allocated:

poolReward -> 25
newRedeemable -> 100
newBonded -> 75

Reply:
@emptysetsquad The 20% is also applied below to the excess given to bonded, so the net result is 20% regardless of expansion/contraction. That was the intention behind how I coded it and unless there is a logic bug. Were we going for something else?